packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -500,11 +500,15 @@ export class Function extends FunctionBase {
         resourceName: functionName,
         arnFormat: ArnFormat.COLON_RESOURCE_NAME,
       }),
+      sameEnvironment: true,
     });
   }
 
   /**
-   * Import a lambda function into the CDK using its ARN
+   * Import a lambda function into the CDK using its ARN.
+   *
+   * To be able to add permissions to this imported lambda, make sure that is in the same account
+   * and region as the stack you are importing it into.
```

Comment:
Maybe phrase this as `Function.addPermissions()` specifically, instead of "add permissions", because it depends on how you do it.

I think we also need to be a bit more specific on when it works and when it doesn't. Use the blurb I typed for some inspiration.